### PR TITLE
Test new RCS with anonymous users

### PR DESCRIPTION
**Pull request by ywangd**
This PR adds tests that show anonymous users work as intended in the new RCS setup.

### CODE REVIEW

**Comment by ywangd** [comment]
For context, the REST test was migrated to use the latest REST test plugin in #93677 
So the code might look very different from what it was yesterday. But it is incremental from #93677

**Review by n1v0lg — APPROVED** [review]
LGTM

File: x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/AbstractRemoteClusterSecurityTestCase.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    protected static ElasticsearchCluster queryCluster;
+    protected static RestClient fulfillingClusterClient;
+
+    @Before
+    public void initFulFillingClusterClient() throws IOException {
+        if (fulfillingClusterClient == null) {
+            assert fulfillingCluster != null;
+            fulfillingClusterClient = buildClient(fulfillingCluster.getHttpAddresses());
+        }
+        assert fulfillingClusterClient != null;
```
[the remark this answers is not in the document]
Comment (ywangd) [reply x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/AbstractRemoteClusterSecurityTestCase.java]: I don't think so. It was copied from elsewhere ... I removed it.

File: x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/AbstractRemoteClusterSecurityTestCase.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+public abstract class AbstractRemoteClusterSecurityTestCase extends ESRestTestCase {
+
+    protected static final String USER = "test_user";
+    protected static final SecureString PASS = new SecureString("x-pack-test-password".toCharArray());
+    protected static final String REMOTE_SEARCH_USER = "remote_search_user";
+    protected static final String REMOTE_SEARCH_ROLE = "remote_search";
+
+    protected static LocalClusterConfigProvider commonClusterConfig = cluster -> cluster.module("analysis-common")
+        .feature(FeatureFlag.NEW_RCS_MODE)
```
Comment (ywangd) [inline x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/AbstractRemoteClusterSecurityTestCase.java:39]: This is where the feature flag is specified.

File: x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/AbstractRemoteClusterSecurityTestCase.java [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+import org.elasticsearch.test.cluster.util.resource.Resource;
+import org.elasticsearch.test.rest.ESRestTestCase;
+import org.junit.AfterClass;
+import org.junit.Before;
+
+import java.io.IOException;
+import java.util.Map;
+
+public abstract class AbstractRemoteClusterSecurityTestCase extends ESRestTestCase {
+
```
[the remark this answers is not in the document]
Comment (ywangd) [reply x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/AbstractRemoteClusterSecurityTestCase.java:32]: See my comment below https://github.com/elastic/elasticsearch/pull/93678/files#r1102764722

**Pull request merged**